tests/Doctrine/Tests/DBAL/Functional/WriteTest.php

```diff
@@ -135,20 +138,46 @@ public function testUpdate() : void
 
     public function testLastInsertId() : void
     {
-        if (! $this->connection->getDatabasePlatform()->prefersIdentityColumns()) {
```

Comment:
This `prefersIdentityColumns()` method is indeed confusing and it's not used anywhere except for tests. We should probably get rid of it (out of scope).

Reply:
I trust you on this one ;)